config/src/config/indexer_table_info_config.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::config::{
+    config_sanitizer::ConfigSanitizer, node_config_loader::NodeType, Error, NodeConfig,
+};
+use aptos_types::chain_id::ChainId;
+use serde::{Deserialize, Serialize};
+
+// Useful defaults
+pub const DEFAULT_PARSER_TASK_COUNT: u16 = 20;
+pub const DEFAULT_PARSER_BATCH_SIZE: u16 = 1000;
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
+#[serde(default, deny_unknown_fields)]
+pub struct IndexerTableInfoConfig {
+    pub enabled: bool,
+
+    /// Number of processor tasks to fan out
+    pub parser_task_count: u16,
+
+    /// Number of transactions each parser will process
+    pub parser_batch_size: u16,
+}
+
+// Reminder, #[serde(default)] on IndexerTableInfoConfig means that the default values for
+// fields will come from this Default impl, unless the field has a specific
+// #[serde(default)] on it (which none of the above do).
+impl Default for IndexerTableInfoConfig {
+    fn default() -> Self {
+        Self {
+            enabled: false,
+            parser_task_count: DEFAULT_PARSER_TASK_COUNT,
+            parser_batch_size: DEFAULT_PARSER_BATCH_SIZE,
+        }
+    }
+}
+
+impl ConfigSanitizer for IndexerTableInfoConfig {
+    fn sanitize(
+        node_config: &NodeConfig,
+        _node_type: NodeType,
+        _chain_id: Option<ChainId>,
+    ) -> Result<(), Error> {
+        let sanitizer_name = Self::get_sanitizer_name();
+
+        if !node_config.indexer_table_info.enabled {
```

Comment:
I'm not sure I follow how this is meant to work. The default for `IndexerTableInfoConfig` is that `enabled` will be false, right? But the sanitizer is going to fail if the config is not enabled? So, any nodes that don't run with the indexer will fail this check and not be able to start up?

My feeling is you likely don't need this check at all? But, if you do need it, you need to call it here: https://github.com/aptos-labs/aptos-core/blob/b17fda2d344ec42ff52eea9fbab7a097524a3c7e/config/src/config/config_sanitizer.rs#L44